File: selenium-webdriver-nodejs-mocha/spec/flight_spec.js
var webdriver = require('selenium-webdriver'),
                By = webdriver.By,
                until = webdriver.until;
var test = require('selenium-webdriver/testing');
var assert = require('assert');

var driver;
const timeOut = 15000;

String.prototype.contains = function(it) { return this.indexOf(it) != -1; };

test.describe('Flight', function () {

  test.before(function() {
     this.timeout(timeOut);
      driver = new webdriver.Builder()
          .forBrowser('chrome')
          .build();
		 driver.get('http://travel.agileway.net');
		 driver.findElement(webdriver.By.name('username')).sendKeys('agileway');
		 driver.findElement(webdriver.By.name('password')).sendKeys('testwise');
		 driver.findElement(webdriver.By.name('commit')).click();
  });

  test.beforeEach(function() {
    this.timeout(timeOut);
    driver.get('http://travel.agileway.net');
  });

  test.after(function() {
    driver.quit();
  });

  test.it('[3] Return trip', function() {
	 driver.findElement(By.xpath("//input[@name='tripType' and @value='return']")).click();
	 driver.findElement(By.name("fromPort")).sendKeys("Sydney");
	 driver.findElement(By.name("toPort")).sendKeys("New York");
	 driver.findElement(By.name("departDay")).sendKeys("02");
	 driver.findElement(By.name("departMonth")).sendKeys("May 2016");
	 driver.findElement(By.name("returnDay")).sendKeys("04");
	 driver.findElement(By.name("returnMonth")).sendKeys("June 2016");	 
	 driver.findElement(By.xpath("//input[@value='Continue']")).click();
	 driver.findElement(By.tagName("body")).getText().then(function(the_page_text){
	   assert(the_page_text.includes("2016-05-02 Sydney to New York"))
	   assert(the_page_text.includes("2016-06-04 New York to Sydney"))
	 });
  });

  test.it.only('[2] One-way trip', function() {
	 driver.findElement(By.xpath("//input[@name='tripType' and @value='oneway']")).click();	
	 driver.findElement(By.name("fromPort")).sendKeys("New York");
	 driver.findElement(By.name("toPort")).sendKeys("Sydney");
	 driver.findElement(By.name("departDay")).sendKeys("02");
	 driver.findElement(By.name("departMonth")).sendKeys("May 2016");
	 driver.findElement(By.xpath("//input[@value='Continue']")).click();
	
	 driver.findElement(By.tagName("body")).getText().then(function(the_page_text){
	   assert(the_page_text.includes("2016-05-02 New York to Sydney"))
	 });
  });

});

File: pre-refactoring/spec/flight_spec.rb
load File.dirname(__FILE__) + '/../test_helper.rb'

describe "Select Flights" do
  include TestHelper

  before(:all) do
    @driver = $browser = Selenium::WebDriver.for(browser_type, browser_options)
    @driver.navigate.to(site_url) # you may

    driver.find_element(:id, "username").send_keys("agileway")
    driver.find_element(:id, "password").send_keys("testwise")
    driver.find_element(:xpath,"//input[@value='Sign in']").click
  end

  after(:all) do
    @driver.quit unless debugging?
  end

  before(:each) do
    # before each test, make sure on flight page
    driver.navigate.to "#{site_url}/flights/start" 
  end

  it "One-way trip" do
    driver.find_element(:xpath, "//input[@name='tripType' and @value='oneway']").click
    expect(driver.find_element(:id, "returnTrip").displayed?).to be_falsey
    Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "fromPort")).select_by(:text, "Sydney")
    Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "toPort")).select_by(:text, "New York")
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "departDay")).select_by(:text, "02")
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "departMonth")).select_by(:text, "May 2016")
    driver.find_element(:xpath,"//input[@value='Continue']").click

    expect(page_text).to include("2016-05-02 Sydney to New York")
  end
  
  it "Return trip" do
    driver.find_element(:xpath, "//input[@name='tripType' and @value='return']").click

    Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "fromPort")).select_by(:text, "Sydney")
    Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "toPort")).select_by(:text, "New York")
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "departDay")).select_by(:text, "02")
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "departMonth")).select_by(:text, "May 2016")
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "returnDay")).select_by(:text, "04")
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "returnMonth")).select_by(:text, "June 2016")
    driver.find_element(:xpath,"//input[@value='Continue']").click

    expect(page_text).to include("2016-05-02 Sydney to New York")
    expect(page_text).to include("2016-06-04 New York to Sydney")
  end

end
